### PR TITLE
Remove v prefix from a number of VectorUtils functions

### PR DESCRIPTION
**Pull request by blaxill**
I think I named these, but now I don't think these need to be prefixed. Where there is ambiguity with list functions, the module name could/should be used to differentiate. Also simplifies the traverse definition.

### CODE REVIEW

**Review by jadephilipoom — APPROVED** [review]
LGTM modulo a naming nitpick. I have some preference for the module name to match the standard library module name; that way I don't have to remember where a function is defined to call it. This could be solved by either creating a `Vector` module inside `VectorUtils.v`, or renaming `VectorUtils.v` to `Vector.v` (or better yet, `Util/Vector.v`).

**Comment by blaxill** [comment]
> (or better yet, `Util/Vector.v`).

I think this may create an extraction issue- we were doing this kind of thing before but we ended up with Haskell like `Vector.hs` `Vector0.hs` etc. I'll experiment with it and also using a `Vector` module inside the file, before merging.

**Comment by blaxill** [comment]
After investigating a little: Due to our extraction setup, using a `Vector.v` filename generates a `Vector0.hs` haskell filename due to the standard library collision, using a `Module Vector` inside the file makes extracted Haskell function names be prefixed with `_Vector__namehere`. I'm going to merge the PR as is, with the idea of fixing the extraction side of this sometime through more intelligent extraction (see #50), and then at the point unify the names

**Pull request merged**
